Block logging service in sandbox

pvollan opened the pull request:
#### 2cf062084175ea94ebfc01afc43709856aa81587
<pre>
Block logging service in sandbox
<a href="https://bugs.webkit.org/show_bug.cgi?id=252195">https://bugs.webkit.org/show_bug.cgi?id=252195</a>
rdar://105416416

Reviewed by Brent Fulgham.

Block logging service in the WebContent process sandbox unless the build is internal. This requires
some tweaks at startup of the WebContent process to avoid opening up a Mach connection to the service
before determining if logging should be enabled. Logs will be forwarded to the Networking process for
logging there if logging is disabled in the WebContent process.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/spi/cocoa/OSLogSPI.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::logOnBehalfOfWebContent):
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.messages.in:
* Source/WebKit/Platform/Logging.h:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceMain.mm:
(WebKit::initializeCFPrefs):
(WebKit::blockLogdInSandbox):
(WebKit::initializeLogd):
(WebKit::XPCServiceEventHandler):
(WebKit::XPCServiceMain):
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::ProcessLauncher::launchProcess):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformInitializeWebProcess):
(WebKit::registerLogHook):
(WebKit::WebProcess::platformInitializeProcess):
(WebKit::initializeLogd): Deleted.
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: <a href="https://commits.webkit.org/261045@main">https://commits.webkit.org/261045@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9002cc0fdafa3800b6c40819e39af69ef537057b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/110392 "4 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/19480 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/42995 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1776 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/119316 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/20920 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/10637 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/102635 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/116135 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/15584 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/98773 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/43801 "Found 2 new test failures: webgl/2.0.0/conformance2/textures/image_bitmap_from_video/tex-2d-r8ui-red_integer-unsigned_byte.html, webgl/2.0.0/conformance2/textures/webgl_canvas/tex-2d-r11f_g11f_b10f-rgb-float.html (failure)") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/113813 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/97545 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30438 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/85660 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/99115 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12145 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31776 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/100127 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/10207 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/12728 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/8745 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/31251 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18093 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/51391 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/108150 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7646 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/14583 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26659 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

smfr reviewed (changes requested):
It seems like this will prevent useful logging for hard-to-reproduce bugs where the user hasn't opened Feedback Assistant before.

brentfulgham commented on Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
```diff
@@ -1431,6 +1431,32 @@ void NetworkConnectionToWebProcess::installMockContentFilter(WebCore::MockConten
+
+    for (unsigned i = 0; i < stringLength; i++) {
+        char c = logString[i];
+        if (c >= 32 && c <= 126)
```
This character validation is used in a few places in WebKit code. Might be nice to make a helper function someday.

pvollan replied:
I will see if I can identify the other places this is done, and look into creating a helper function.

brentfulgham commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -457,10 +457,8 @@
-    (global-name "com.apple.logd.events")
-)
+(with-filter (require-not (state-flag "DisableLogging"))
+    (allow mach-lookup (global-name "com.apple.logd" "com.apple.logd.events")))
```
Nit: Showing each service on its own line is more readable/easier to review.

pvollan replied:
Fixed!

brentfulgham reviewed (approved):
I think this looks good. Should we do any A/B testing to confirm this doesn't impact launch performance?

pvollan committed:
Block logging service in sandbox
https://bugs.webkit.org/show_bug.cgi?id=252195
rdar://105416416

Reviewed by Brent Fulgham.

Block logging service in the WebContent process sandbox unless the build is internal. This requires
some tweaks at startup of the WebContent process to avoid opening up a Mach connection to the service
before determining if logging should be enabled. Logs will be forwarded to the Networking process for
logging there if logging is disabled in the WebContent process.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/spi/cocoa/OSLogSPI.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::logOnBehalfOfWebContent):
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.messages.in:
* Source/WebKit/Platform/Logging.h:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceMain.mm:
(WebKit::initializeCFPrefs):
(WebKit::blockLogdInSandbox):
(WebKit::initializeLogd):
(WebKit::XPCServiceEventHandler):
(WebKit::XPCServiceMain):
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::ProcessLauncher::launchProcess):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformInitializeWebProcess):
(WebKit::registerLogHook):
(WebKit::WebProcess::platformInitializeProcess):
(WebKit::initializeLogd): Deleted.
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: https://commits.webkit.org/261045@main
